GitUI/UserControls/RevisionGridClasses/DvcsGraph.cs

```diff
@@ -582,6 +594,7 @@ private void BackgroundThreadEntry()
             }
         }
 
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2002:DoNotLockOnObjectsWithWeakIdentity")]
```

Comment:
:bulb: I'm not convinced this is correct. I would prefer to see it moved to its own pull request and evaluated separately.